pkgdb/tests/buildenv.bats

```diff
@@ -61,6 +61,23 @@ setup_file() {
   assert_success
 }
 
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=unfree,unfree:fail
+@test "Environment with unfree packages fails to build by default" {
+  run "$PKGDB_BIN" buildenv "$LOCKFILES/unfree/manifest.lock"
+  assert_failure
+  assert_output --partial "'hello' is marked as unfree"
+}
+
+# bats test_tags=unfree,unfree:success
+@test "Environment with unfree packages succeeds if allwed in options" {
+  run "$PKGDB_BIN" buildenv "$LOCKFILES/unfree-allowed/manifest.lock"
+  assert_success
+}
+
+# ---------------------------------------------------------------------------- #
+
```

Comment:
suggestion blocking: add coverage for broken

Reply:
@ysndr ping on this, just want to double check you aren't waiting for me on this PR

Reply:
I opened https://github.com/flox/flox/pull/1099

To track the verification of lockfiles, I don't think we should check the same things multiple times in different phases and would migrate these tests over to the new PR.
I dont know if you'd want to wait merging this until after #1099 or if we can move things in the follow up.

This PR also conflicts with #1094 in parts, do we want to prioritize either @mkenigs @zmitchell

Reply:
No preference on my end